docs/source/developers/guide/step_by_step/set_up.rst

```diff
@@ -27,3 +27,98 @@
+operating systems.
+
+You can follow the instructions to install git from GitHub
+where Arrow repository is hosted, following
+`the quickstart instructions <https://docs.github.com/en/get-started/quickstart/set-up-git>`_.
+
+When git is set up do not forget to configure your name and email
+
+.. code:: console
+
+   $ git config --global user.name "Your Name"
+   $ git config --global user.email your.email@example.com
+
+and `authenticate with GitHub <https://docs.github.com/en/get-started/quickstart/set-up-git#next-steps-authenticating-with-github-from-git>`_
+as this will allow you to interact with GitHub without typing
+a username and password each time you execute a git command.
+
+**The Terminal**
+
+You might be able to get away without using the command-line for
+most tasks, but for some it seems to be the most straightforward way.
+
+Get the source code
+===================
+
+Fork the repository
+-------------------
+
+The Arrow GitHub repository contains both the Arrow C++ library and
+the various bindings to it (libraries in Python, R, Rust, Java, etc.).
+The first step to contributing is to create a fork of the repository
+in your own GitHub account.
+
+1. Go to `<https://github.com/apache/arrow>`_.
+
+2. Press Fork in the top right corner.
+
+   .. figure:: github_fork.jpeg
+      :scale: 50 %
+      :alt: Fork the Apache Arrow repository on GitHub.
+
+      The icon to fork the Apache Arrow repository on GitHub.
+
+3. Choose to fork the repository to your username so the fork will be
+   created at ``https://github.com/YOU/arrow``.
+
+Clone the repository
+--------------------
+
+Next you need to clone the repository (this is where the command line
+comes in). It is *highly recommended* to clone the repository using
+the command line or a Git client. Cloning it in an IDE might create
+unnecessary headaches - for example, cloning it in RStudio assumes the
+whole repository is an R project and will create the ``.Rproj`` file
+in the root directory.
+
+.. code:: console
+
+   $ git clone https://github.com/YOU/arrow.git
+   $ cd arrow
+   $ git remote add upstream https://github.com/apache/arrow
+
+Verify your upstream
+--------------------
+
+Your upstream should be pointing at the Arrow GitHub repo.
+
+Running in the shell:
+
+.. code:: console
+
+   $ git remote -v
+
+Should give you a result similar to this:
+
+.. code:: console
+
+   origin	https://github.com/YOU/arrow.git (fetch)
+   origin	https://github.com/YOU/arrow.git (push)
+   upstream	https://github.com/apache/arrow (fetch)
+   upstream	https://github.com/apache/arrow (push)
+
+If you did everything correctly, you should now have a copy of the code
+in the Arrow directory and two remotes that refer to your own GitHub
+fork (``origin``) and the official Arrow repository (``upstream``).
+
+Note that in the official documentation for Arrow the folder is sometimes
+shown to be cloned into the repository folder. That is not mandatory and
```

Comment:
Hmm, I think I don't understand what this means. What is "the repository folder"? I'm a bit confused...

Reply:
I added this due to the instructions for [Python Development](https://arrow.apache.org/docs/developers/python.html#environment-setup-and-build) suggesting to add in `Arrow` into `repos` folder.

Reply:
Ah! I would rather remove the `mkdir repos` and `cd repos` there entirely. People organize their directories as they want - I don't have a `repos` folder, for example, and I'm sure I'm not the only one :-)

Reply:
OK, will do all in this PR if u agree.

Reply:
Please do!